craft_providers/bases/almalinux.py

```diff
@@ -0,0 +1,844 @@
+#
+# Copyright 2021-2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""Almalinux image(s)."""
+import enum
+import io
+import logging
+import pathlib
+import re
+import subprocess
+import sys
+import time
+from datetime import datetime, timedelta
+from time import sleep
+from typing import Dict, List, Optional, Type
+
+from pydantic import ValidationError
+
+from craft_providers import Base, Executor, errors
+from craft_providers.actions import snap_installer
+from craft_providers.actions.snap_installer import Snap, SnapInstallationError
+from craft_providers.errors import BaseCompatibilityError, BaseConfigurationError
+from craft_providers.util.os_release import parse_os_release
+
+from .instance_config import InstanceConfiguration
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=duplicate-code
+
+
+class AlmaLinuxBaseAlias(enum.Enum):
+    """Mappings for supported AlmaLinux images."""
+
+    NINE = "9"
+
+
+class AlmaLinuxBase(Base):
```

Comment:
How different is this from the CentOS base?

Reply:
Probably about 95% are the same, so there's another PR to do a overall refactor.